src/test/java/teammates/sqlui/webapi/GetOngoingSessionsActionTest.java

```diff
@@ -0,0 +1,417 @@
+        allOngoingSessions.add(c2Fs1);
+        allOngoingSessions.add(c3Fs1);
+        when(mockDatastoreLogic.getAllOngoingSessions(start, end)).thenReturn(allOngoingSessions);
+
+        long startTime = start.toEpochMilli();
+        long endTime = end.toEpochMilli();
+        String startTimeString = String.valueOf(startTime);
+        String endTimeString = String.valueOf(endTime);
+        String[] params = {
+                Const.ParamsNames.FEEDBACK_SESSION_STARTTIME, startTimeString,
+                Const.ParamsNames.FEEDBACK_SESSION_ENDTIME, endTimeString,
+        };
+
+        GetOngoingSessionsAction getOngoingSessionsAction = getAction(params);
+        JsonResult r = getJsonResult(getOngoingSessionsAction);
+        OngoingSessionsData response = (OngoingSessionsData) r.getOutput();
+
+        assertEquals(3, response.getTotalOngoingSessions());
+        assertEquals(1, response.getTotalOpenSessions());
+        assertEquals(1, response.getTotalClosedSessions());
+        assertEquals(1, response.getTotalAwaitingSessions());
+        assertEquals(3L, response.getTotalInstitutes());
+        Map<String, List<OngoingSession>> expectedSessions = new HashMap<>();
+        OngoingSession expectedOngoingC1Fs2 = new OngoingSession(sqlC1Fs2, instructor2.getGoogleId());
+        expectedSessions.put("NUS", Collections.singletonList(expectedOngoingC1Fs2));
+        OngoingSession expectedOngoingC2Fs1 = new OngoingSession(sqlC2Fs1, instructor3.getGoogleId());
+        expectedSessions.put("MIT", Collections.singletonList(expectedOngoingC2Fs1));
+        OngoingSession expectedOngoingC3Fs1 = new OngoingSession(c3Fs1, instructor4.getGoogleId());
+        expectedSessions.put("UCL", Collections.singletonList(expectedOngoingC3Fs1));
+        Map<String, List<OngoingSession>> actualSessions = response.getSessions();
+        assertEqualSessions(expectedSessions, actualSessions);
+    }
+
+    @Test
+    void testExecute_courseMigratedButAccountNotMigrated_shouldGetOngoingSessionsDataCorrectly() {
+        // The Instant input parameters into the mock methods have a precision up to the nanoseconds, but the time
+        // input parameters into the Action only have a precision up to the milliseconds. We must truncate to
+        // milliseconds so that the mock methods can mock the exact time that the Action would parse, instead of
+        // mocking a time that is off by an amount of time less than a millisecond.
+        Instant instantNow = Instant.now().truncatedTo(ChronoUnit.MILLIS);
+        Instant start = instantNow.minus(Duration.ofDays(1L));
+        Instant end = instantNow.plus(Duration.ofDays(1L));
+        Course sqlCourse2 = new Course("test-id2", "test-name2", "UTC", "MIT");
+        when(mockLogic.getCourse(sqlCourse2.getId())).thenReturn(sqlCourse2);
+        // Account instructor3Account = new Account("instructor3", "instructor3", "test3@test.com");
```

Comment:
nit: remove this comment

Reply:
Oops, I must have missed these. Thanks for spotting them.